src/main/java/de/unistuttgart/t2/common/scaling/AutoscalingController.java

```diff
@@ -15,53 +20,109 @@
 @RestController
 public class AutoscalingController {
 
+    private static final Logger logger = LoggerFactory.getLogger(AutoscalingController.class);
+
     @Operation(summary = "Unblock all requests", description = "Lifts the block for any route", tags = "Block")
     @ApiResponses(value = @ApiResponse(responseCode = "200", description = "Successfully lifted the route block"))
     @PostMapping("/autoscaling/unblock-routes")
     public void unblockRoutes() {
         RequestDenier.shouldBlockAllRoutes(false);
+        logger.info("Unblocked all non-autoscaling routes");
     }
 
     @Operation(summary = "Block all subsequent requests", description = "deterministically produces a SLO for all subsequent requests outside of \"/autoscaling/*\" ", tags = "Block")
     @ApiResponses(value = @ApiResponse(responseCode = "200", description = "Successfully blocked all further routes"))
     @PostMapping("/autoscaling/block-routes")
     public void blockRoutes() {
         RequestDenier.shouldBlockAllRoutes(true);
+        logger.warn("Blocked all non-autoscaling routes");
     }
 
-    @Operation(summary = "Ensures that consistently at least {memory}% is used", description = "values in range 0 < {memory} < 1 are treated the same as values in range 1 <= {memory} < 100", tags = "Memory")
+    @Operation(summary = "Ensures that consistently at least {memory}% is used", description = "values in range 0 <= {memory} < 1 are treated the same as values in range 1 <= {memory} < 100", tags = "Memory")
     @ApiResponses(value = {
         @ApiResponse(responseCode = "200", description = "Successfully demanded a memory usage of at least {memory}%"),
-        @ApiResponse(responseCode = "400", description = "{memory} >= 100.0") })
+        @ApiResponse(responseCode = "400", description = "{memory} >= 100.0 || {memory} < 0.0") })
     @PostMapping("/autoscaling/require-memory/{memory}")
-    public void requireMemory(@PathVariable(name = "memory") double memory) {
-        try {
-            MemoryLeaker.changeExpectedMemoryPercentage(memory);
-        } catch (IllegalArgumentException e) {
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, String
-                .format("%f is greater than the allowed maximum value of 100.0", memory));
-        }
+    @ResponseBody
+    public MemoryInfo requireMemory(@PathVariable(name = "memory") double memory) {
```

Comment:
Okay, yeah, might be possible, I simply thought it would make sense not to have any conversion when not necessary...

Reply:
What might be possible and easy to implement: `/autoscaling/require-memory/{memory}-percent`.
But I do not see really see the benefit over the current `/autoscaling/require-memory/{memory}`...

Reply:
Improved the comment now.

Reply:
> What might be possible and easy to implement: `/autoscaling/require-memory/{memory}percent`. But I do not see really see the benefit over the current `/autoscaling/require-memory/{memory}`...

This looks like a good solution to me. For me, the main benefit would be a clearer contract. Imagine someone builds a UI based on this, and adds a slider from 0 to 100. The first easy way to implement this would be just calling the api with the value. But this will result in strange bugs, because as soon as the user chooses a value below 1 (maybe even wants zero, but misses it a little bit), way more load is put on the cpu than the user expects. Basically, the current implementation is nice to use manually, but its contract allows for bugs to happen way more easily than necessary.

Reply:
And what I just realized, the other way to implement it (slider from 0 to 1) would also be buggy, as when the user selects 1, the system now interprets it as 0.01

Reply:
Implemented now.